### ISSUE TITLE
[Feature request] Button remap

### ISSUE THREAD

**Issue by Gaboros** [ndeadly/MissionControl#209]
Some more advanced button remap than official would be great. Currently I can not map screenshot button on my Xbox One controller. 

Mapping not "official" buttons would be even better. :)

For example on sys-con I can make a screenshot with the sync button.

**Comment by ndeadly** [comment]
Are you aware of the button combos for `home` and `capture` buttons? I've assigned the combos `MINUS + DPAD_UP` (capture) and `MINUS + DPAD_DOWN` (home) for all controllers to combat this issue where there are no spare buttons left on some controllers for these buttons.

I'm a bit on the fence about opening up full control over the button mapping at the moment. It would be nice to have, but it adds a bunch of additional complexity to the button processing. I like the idea of keeping it as simple and fast as possible to minimise any added input delay. It's something I might consider down the road once other major features are implemented, but for now I think the builtin HOS remapping covers most people's needs. I do have plans to add a few other unsupported mapping options in the meantime though, such as swapping dpad and left stick and inverting stick y-axis.

I wasn't aware of the sync button state being sent in HID reports. I just checked on my controller and it doesn't look like it gets sent over Bluetooth. Must be USB only. In general I map every button on a controller where possible unless there are leftovers that aren't needed.
